### PR TITLE
Extending API

### LINKED ISSUE #503: extending API
# Problem

The API needs to be reviewed to a point where it is fully pluggable.

# Proposed Solution

Investigate which API views are missing and add them.

### PR DESCRIPTION
**Pull request by sonlinux**
Fix #503 
I have added most of the missing API views to the existing API, which include for models:

- address

-  education training quality

- abet band

-  national graduate school in education

- national certificate vocation

- feedback

- qual class

### CODE REVIEW

**Comment by sonlinux** [comment]
@ann26 could you please take a quick review on this PR before I merge it, thanks.

**Comment by meomancer** [comment]
hi @sonlinux 
i'm not sure the purpose of the API that just return all
my suggestion is maybe API can be filtered by some fields
or maybe you could change url using
```
api/national_graduate-school_in_education/all
```
because in my mind, API can be filtering, not just return all data, because it will not be usable

File: django_project/feti/api_views/abet_band.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+from feti.serializers.abet_band_serializer import AbetBandSerializer
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '22/10/17'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+class AbetBandApiView(APIView):
```
Comment (anitanh) [inline django_project/feti/api_views/abet_band.py]: hi @sonlinux what is this?

File: django_project/feti/api_views/address.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+__copyright__ = 'kartoza.com'
+
+
+class AddressApiView(APIView):
+    """
+    Api to list all the Addresses.
+    """
+
+    def get(self, request):
+        bands = Address.objects.all()
```
Comment (meomancer) [inline django_project/feti/api_views/address.py]: please rename variable accordingly

**Review by anitanh — CHANGES_REQUESTED** [review]

File: django_project/feti/api_views/address.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from feti.serializers.address_serializer import AddressSerializer
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '22/10/17'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+class AddressApiView(APIView):
```
Comment (anitanh) [inline django_project/feti/api_views/address.py:14]: Returning only all addresses are meaningless, we don't know whose address is this

File: django_project/feti/api_views/feedback.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+from feti.serializers.feedback_serializer import FeedbackSerializer
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '22/10/17'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+class FeedbackApiView(APIView):
```
Comment (anitanh) [inline django_project/feti/api_views/feedback.py]: I am not sure if we need this. We can see all feedbacks in the admin view though.

**Comment by anitanh** [comment]
I agree with @meomancer

File: django_project/feti/api_views/address.py (hunk 2 above)
Comment (sonlinux) [reply django_project/feti/api_views/address.py]: Alright thanks @meomancer let me do so.

**Comment by sonlinux** [comment]
Thanks @meomancer and @ann26 let me make the necessary changes.

**Comment by dimasciput** [comment]
@sonlinux any progress for this?

**Pull request closed without merging**